### ISSUE TITLE
Multi colour OS images

### ISSUE THREAD

**Issue by FlyinPancake** [LivacoNew/CrabFetch#31]
**What would you like to see added to CrabFetch?**
I'd love to have multi colour os images

**What use would this be?**
Colourful images are much prettier to look at :)
Feature compatibility with fastfetch would be also great as I try to oxidize my sw stack 

**Additional Comments**

So pretty
![image](https://github.com/user-attachments/assets/eedae20b-bc84-44b0-9bcf-887eda8accd0)

Not so pretty
![image](https://github.com/user-attachments/assets/066c28d4-0cb2-4cc1-bad3-d84c3741e72e)

**Comment by LivacoNew** [comment]
Gonna see if I can rework this into a full redo of the ASCII coloring system, as it's too limiting for this to be a reality for now

**Comment by FlyinPancake** [comment]
Yes, I've seen that is is simplistic, just wanted to add a tracking issue

**Comment by LivacoNew** [comment]
@FlyinPancake Mind trying out the [better-ascii](https://github.com/LivacoNew/CrabFetch/tree/better-ascii) branch? It should let you do what you want using the `raw` mode, while also redoing how the ASCII's are processed to be better.

**Comment by LivacoNew** [comment]
If it's letting you do what you want lemme know and I'll merge it into main :)

**Comment by FlyinPancake** [comment]
Can you help out a bit with the config? I'm a bit lost hadn't had to touch it in ages

**Comment by FlyinPancake** [comment]
okay so ghostty is detected a bit wierd: 

![Image](https://github.com/user-attachments/assets/e17f8fa3-14ce-4a61-9fee-f8e772843007)

**Comment by LivacoNew** [comment]
> okay so ghostty is detected a bit wierd:
> 
> ![Image](https://github.com/user-attachments/assets/e17f8fa3-14ce-4a61-9fee-f8e772843007)

wtf lmao, can't replicate this, mind doing some more testing to see where those newlines and "Version" is coming from? e.g a specific module or whatever

![Image](https://github.com/user-attachments/assets/90c6f60d-1739-4661-9ec7-317b730efdcf)


> Checked it with an easier example:
> 
> ![Image](https://github.com/user-attachments/assets/8988626a-e05b-4a98-9d7f-7a0d403ac35c)
> 
> I imagine this will work, i'll just make a quick converter for fastfetch logos :D

I'll wait and see how you get on with an actual logo before merging just incase

**Comment by LivacoNew** [comment]
> Can you help out a bit with the config? I'm a bit lost hadn't had to touch it in ages

You're looking for these settings

![Image](https://github.com/user-attachments/assets/efb19e95-20c8-42ea-bb11-74335a8dc1c5)

Use the "raw" mode w/ an `.config/CrabFetch/ascii` file that has your logo w/ ANSI color codes

**Comment by FlyinPancake** [comment]
Punched out the converter code :D still a bit buggy tho

![Image](https://github.com/user-attachments/assets/4f4416ff-c750-4141-8fae-de74c4a8b9d6)

**Comment by FlyinPancake** [comment]
I'll have to reset the colors at the end of the line and set the previous one on the beginnings do i 

![Image](https://github.com/user-attachments/assets/87bfa00a-adfd-4f39-9042-43392c0dccf0)

**Comment by FlyinPancake** [comment]
![Image](https://github.com/user-attachments/assets/e25b4804-8f0e-4c67-9902-cfedff3f976e)

Just checked if I'd accidentally inserted a bunch of whitespace on the ends of the line, but no. It appears that the spacing needs a second look.

**Comment by FlyinPancake** [comment]
[ascii.tar.gz](https://github.com/user-attachments/files/18804076/ascii.tar.gz)

Here's my logo gztared to pass GH's mime reqs

**Comment by LivacoNew** [comment]
okay, try this patch: [patch.txt](https://github.com/user-attachments/files/18804153/patch.txt)

what's happening is the ansi codes are being added to the character count that the spacing code uses to pad the ascii out, this should fix it

**Comment by FlyinPancake** [comment]
![Image](https://github.com/user-attachments/assets/85a49fe0-8f61-4f8d-aeaa-bd052f931d1e)

**Comment by FlyinPancake** [comment]
looks good!

i'm fixing the converter code and i'll push it to gh as soon as it's done

**Comment by LivacoNew** [comment]
I won't bother with the conversion code, although your free to put it in another repo of your own if you want of course :)

**Comment by LivacoNew** [comment]
Okay, thank you much better. Will merge to main now

**Comment by FlyinPancake** [comment]
Fixed the conv code now looks epic: 

![Image](https://github.com/user-attachments/assets/7839aec7-d478-4f1e-a087-26a96c5f0ad5)

Here's the converter code repo, standalone binary might be helpful to someone in the future: https://github.com/FlyinPancake/fastfetch2ascii

**Comment by LivacoNew** [comment]
PR has been merged, thanks for the help in testing :)
